modules/porous_flow/src/materials/PorousFlowPermeabilityKozenyCarman.C

```diff
@@ -41,39 +43,25 @@ PorousFlowPermeabilityKozenyCarmanTempl<is_ad>::validParams()
   params.addRangeCheckedParam<Real>(
       "d", "d > 0", "The grain diameter, required for kozeny_carman_fd2");
-  params.addRequiredRangeCheckedParam<Real>("n", "n >= 0", "Porosity exponent");
-  params.addRequiredRangeCheckedParam<Real>("m", "m >= 0", "(1-porosity) exponent");
-  params.addClassDescription(
-      "This Material calculates the permeability tensor from a form of the Kozeny-Carman equation, "
-      "k = k_ijk * A * phi^n / (1 - phi)^m, where k_ijk is a tensor providing the anisotropy, phi "
-      "is porosity, n and m are positive scalar constants and A is given in one of the following "
-      "forms: A = k0 * (1 - phi0)^m / phi0^n (where k0 and phi0 are a reference permeability and "
-      "porosity) or A = f * d^2 (where f is a scalar constant and d is grain diameter.");
+  params.addRangeCheckedParam<Real>(
+      "A", "A > 0", "Kozeny Carman permeability multiplying factor, required for kozeny_carman_A");
+  params.addClassDescription("This Material calculates the permeability tensor from a form of the "
+                             "Kozeny-Carman equation based on the spatially constant initial "
+                             "permeability and porosity or grain size.");
   return params;
 }
 
 template <bool is_ad>
 PorousFlowPermeabilityKozenyCarmanTempl<is_ad>::PorousFlowPermeabilityKozenyCarmanTempl(
     const InputParameters & parameters)
-  : PorousFlowPermeabilityBaseTempl<is_ad>(parameters),
+  : PorousFlowPermeabilityKozenyCarmanBaseTempl<is_ad>(parameters),
     _k0(parameters.isParamValid("k0") ? this->template getParam<Real>("k0") : -1),
     _phi0(parameters.isParamValid("phi0") ? this->template getParam<Real>("phi0") : -1),
     _f(parameters.isParamValid("f") ? this->template getParam<Real>("f") : -1),
     _d(parameters.isParamValid("d") ? this->template getParam<Real>("d") : -1),
-    _m(this->template getParam<Real>("m")),
-    _n(this->template getParam<Real>("n")),
-    _k_anisotropy(parameters.isParamValid("k_anisotropy")
-                      ? this->template getParam<RealTensorValue>("k_anisotropy")
-                      : RealTensorValue(1.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 1.0)),
-    _porosity_qp(this->template getGenericMaterialProperty<Real, is_ad>("PorousFlow_porosity_qp")),
-    _dporosity_qp_dvar(is_ad ? nullptr
-                             : &this->template getMaterialProperty<std::vector<Real>>(
-                                   "dPorousFlow_porosity_qp_dvar")),
-    _dporosity_qp_dgradvar(is_ad ? nullptr
-                                 : &this->template getMaterialProperty<std::vector<RealGradient>>(
-                                       "dPorousFlow_porosity_qp_dgradvar")),
     _poroperm_function(this->template getParam<MooseEnum>("poroperm_function")
-                           .template getEnum<PoropermFunction>())
+                           .template getEnum<PoropermFunction>()),
+    _A(parameters.isParamValid("A") ? this->template getParam<Real>("A") : -1)
 {
```

Comment:
you should error if "A" is provided and you are not using "PoropermFunction::kozeny_carman_A"

Reply:
I added error checks in the constructor for incompatible parameters.